exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_metric_exporter/__init__.py

```diff
@@ -79,90 +80,124 @@ def __init__(
         )
 
     def _translate_data(
-        self, data: Sequence[Metric]
+        self, data: MetricsData
     ) -> ExportMetricsServiceRequest:
-        sdk_resource_scope_metrics = {}
 
-        for metric in data:
-            resource = metric.resource
-            scope_map = sdk_resource_scope_metrics.get(resource, {})
-            if not scope_map:
-                sdk_resource_scope_metrics[resource] = scope_map
+        resource_instrumentation_scope_pb2_scope_metrics = {}
 
-            scope_metrics = scope_map.get(metric.instrumentation_scope)
+        for resource_metrics in data.resource_metrics:
 
-            if not scope_metrics:
-                if metric.instrumentation_scope is not None:
-                    scope_map[metric.instrumentation_scope] = pb2.ScopeMetrics(
+            resource = resource_metrics.resource
+
+            instrumentation_scope_pb2_scope_metrics = (
+                resource_instrumentation_scope_pb2_scope_metrics.get(
+                    resource, {}
+                )
+            )
+
+            if not instrumentation_scope_pb2_scope_metrics:
+                resource_instrumentation_scope_pb2_scope_metrics[
+                    resource
+                ] = instrumentation_scope_pb2_scope_metrics
+
+            for scope_metrics in resource_metrics.scope_metrics:
+
+                instrumentation_scope = scope_metrics.scope
+
+                if instrumentation_scope is None:
+                    instrumentation_scope_pb2_scope_metrics[
+                        None
+                    ] = pb2.ScopeMetrics()
+
+                else:
+                    instrumentation_scope_pb2_scope_metrics[
+                        instrumentation_scope
+                    ] = pb2.ScopeMetrics(
                         scope=InstrumentationScope(
-                            name=metric.instrumentation_scope.name,
-                            version=metric.instrumentation_scope.version,
+                            name=instrumentation_scope.name,
+                            version=instrumentation_scope.version,
                         )
                     )
-                else:
-                    scope_map[
-                        metric.instrumentation_scope
-                    ] = pb2.ScopeMetrics()
 
-            scope_metrics = scope_map.get(metric.instrumentation_scope)
+                pb2_scope_metrics = instrumentation_scope_pb2_scope_metrics[
+                    instrumentation_scope
+                ]
+
+                for metric in scope_metrics.metrics:
+                    pb2_metric = pb2.Metric(
+                        name=metric.name,
+                        description=metric.description,
+                        unit=metric.unit,
+                    )
+
+                    for data_point in metric.data.data_points:
+
+                        if isinstance(metric.data, Gauge):
```

Comment:
`metric.data` is one of so it remains same for all the `data_points`. I think you can move this isinstance outside and just use the same pb2 datapoint for the `metric.data.data_points`.

Reply:
Right, fixed :sunglasses: